packages/cli/src/utils/dependencies.js

```diff
@@ -0,0 +1,49 @@
+import consola from 'consola'
+import { satisfies } from 'semver'
+
+import { getPKG } from '@nuxt/utils'
+
+export function checkDependencies() {
+  const packagesToTrack = ['postcss', 'webpack']
+  const webpackDeps = getPKG('@nuxt/webpack').dependencies
+  const packagesWithVersions = Object.entries({ ...webpackDeps }).reduce(
+    (obj, [name, version]) => {
+      if (packagesToTrack.includes(name)) {
+        obj[name] = version
+      }
+      return obj
+    },
+    {}
+  )
+
+  const optionalDependencies = {
+    'sass-loader': '^10.1.1'
+  }
+
+  const versions = {
+    ...optionalDependencies,
+    ...packagesWithVersions
+  }
+
+  Object.entries(versions).forEach(([name, requiredVersion]) => {
+    try {
+      const installedVersion = getPKG(name).version
+      if (!satisfies(installedVersion, requiredVersion)) {
+        consola.error(
+          `Required version of ${name} (${requiredVersion}) not installed. (${installedVersion} was detected.)`
+        )
+      }
+    } catch {
+      if (!Object.keys(optionalDependencies).includes(name)) {
+        consola.warn(`Expected ${name}@${requiredVersion} to be installed.`)
+      }
+    }
+  })
+
+  // Check Node version
+  if (satisfies(process.version, '^10.0.0')) {
+    consola.warn('Node.js 10 is deprecated and will shortly not be supported.')
+  } else if (!satisfies(process.version, '>12.0.0')) {
+    consola.error('Please use a supported version of Node.js (12+).')
```

Comment:
IMO thin can be still a warning for You are using an incompatible NodeJS version. An error is not necessarily happening by using node 10

Reply:
Also no exception will be needed anymore for 10.x (version constraints are always updating)

Reply:
I've moved the <10 to a warn (as 10 is). Not sure what you mean by last comment about 10.x.

Reply:
I mean when `>12.0.0` constraint did not met warning can be generic like _You are using an unsupported version of Node.js (x.y.z) and nuxt is likely to be broken. It is recommended to use latest LTS version (https://nodejs.org)_

note that node@10 is only supported by engine. With 2.15 release notes we are also asking to use 12+. And this might be changed in the future too depending on what happens to nuxt dependencies. So there instead of exception for 10.x, we can have a generic advice and single semver constraint.

Reply:
Very helpful - yes, much better approach. I've made this more generic but happy to refactor further either now or in a further PR.